src/shipit_static_analysis/shipit_static_analysis/clang/__init__.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import tarfile
+import requests
+import io
+import os
+import cli_common.utils
+
+
+def setup(product='static-analysis', job_name='linux64-clang-tidy', revision='latest', artifact='public/build/clang-tidy.tar.xz'):
+    '''
+    Setup Taskcluster clang build for static-analysis
+    Defaults values are from https://dxr.mozilla.org/mozilla-central/source/taskcluster/ci/toolchain/linux.yml
+    - Download the artifact from latest Taskcluster build
+    - Extracts it into the MOZBUILD_STATE_PATH as expected by mach
+    '''
+    namespace = 'gecko.v2.mozilla-central.{}.{}.{}'.format(revision, product, job_name)
+    artifact_url = 'https://index.taskcluster.net/v1/task/{}/artifacts/{}'.format(namespace, artifact)
+
+    # Mach expects clang binaries in this specific root dir
+    target = os.path.join(
+        os.environ['MOZBUILD_STATE_PATH'],
+        'clang-tools',
+    )
+
+    def _download():
```

Comment:
Nit: We could put https://github.com/mozilla-releng/services/blob/master/src/shipit_code_coverage/shipit_code_coverage/taskcluster.py#L81, with some modifications, into cli_common and re-use it both in shipit_code_coverage and shipit_static_analysis.

("Nit" because this is a small piece of code, there isn't a huge benefit in sharing it)

Reply:
I thought about this, but this is so small that i did not think it was worthit